engine-standalone-storage/src/relayer_db/types.rs

```diff
@@ -146,12 +193,47 @@ pub struct TransactionRow {
     pub output: Vec<u8>,
 }
 
-impl From<postgres::Row> for TransactionRow {
-    fn from(row: postgres::Row) -> Self {
-        let block: i64 = row.get("block");
+struct TransactionRowBlock(pub i64);
+
+impl TryFrom<TransactionRowBlock> for u64 {
+    type Error = TryFromIntError;
+
+    fn try_from(value: TransactionRowBlock) -> Result<Self, Self::Error> {
+        // set negative values to 0
+        value.0.max(0).try_into()
```

Comment:
Are you sure it is ok to set the negative value to 0? Is it not better to return an error in case of a negative value?
```suggestion
        value.0.try_into()
```

Reply:
100% agree